traits_futures/wx/pinger.py

```
@@ -0,0 +1,88 @@
+# (C) Copyright 2018-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+"""
+wxPython cross-thread pinger.
+
+This module provides a way for a background thread to request that
+the main thread execute a (fixed, parameterless) callback.
+"""
+
+import wx
+
+
+#: Event type that's unique to this message. (It's just an integer.)
+_PING_EVENT_TYPE = wx.NewEventType()
+
+
+class _PingEvent(wx.PyCommandEvent):
+    """ wx event used to signal that a message has been sent """
+
+    def __init__(self, event_id):
+        wx.PyCommandEvent.__init__(self, _PING_EVENT_TYPE, event_id)
+
+
+class Pinger:
+    """
+    Ping emitter, which can emit pings in a thread-safe manner.
+
+    Parameters
+    ----------
+    pingee : Pingee
+        The corresponding ping receiver.
+    """
+
+    def __init__(self, pingee):
+        self.pingee = pingee
+
+    def connect(self):
+        """
+        Connect to the ping receiver. No pings should be sent until
+        this function is called.
+        """
+        pass
+
+    def disconnect(self):
+        """
+        Disconnect from the ping receiver. No pings should be sent
+        after calling this function.
+        """
+        pass
+
+    def ping(self):
+        """
+        Send a ping to the ping receiver.
+        """
+        event = _PingEvent(-1)
```

Any specific reason why we set `event_id` to `-1`? It doesn't matter what integer value we use, IIUC.
Also, this might be overkill but do we want to raise an error if `event_id` is not an integer? Looks like `wx.PostEvent` is not thread-safe for event objects having string fields.

---

It looks as though I don't actually need this at all. I'll remove it.